getnacelle/nacelle-js
# feat(commerce-queries): add content method

## Why are these changes introduced?

Addresses [ENG-8321](https://nacelle.atlassian.net/browse/ENG-8321) - adds a content method to the commerce query plugin 



## What is this pull request doing?

1. Adds a `withConfig` type to the plugin types in the Storefront sdk to enable getting access to the config in plugins
2. Updates `applyAfter` to be protected instead of private so it's more accessible in the plugins
3. Adds the content method to the plugin
4. Adds a fetchPaginatedData method so that other paginated functions can reuse the same logic

## How to Test

1. Navigate to `packages/storefront-sdk` and run `npm run build && npm link`
2. Navigate to `packages/storefront-sdk-plugin/commerce-queries` and run `npm link @nacelle/storefront-sdk`
3. Build with `npm run build` - should build successfully
5. Test with `npm run test` - all tests should pass
6. Typecheck with `npm run test:typecheck -- --run` - tests and code should all be correctly typed



## Checklist



[ENG-8321]: https://nacelle.atlassian.net/browse/ENG-8321?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
